Esp32_Improv not compiling

### The problem

Esp32-c3 won't compile with esp32_improv


### Which version of ESPHome has the issue?

2023.6.2

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2023.6

### What platform are you using?

ESP32-IDF

### Board

lolin_c3_mini

### Component causing the issue

esp32_improv

### Example YAML snippet

```yaml
esp32_improv:
  authorizer: none
```

### Anything in the logs that might be useful for us?

```txt
Dependency Graph
|-- noise-c @ 0.1.4
|-- Improv @ 1.2.3
Linking /data/esp32-c3/.pioenvs/esp32-c3/firmware.elf
/data/cache/platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/8.4.0/../../../../riscv32-esp-elf/bin/ld: /data/esp32-c3/.pioenvs/esp32-c3/src/esphome/components/esp32_ble/ble_advertising.o: in function `esphome::esp32_ble::BLEAdvertising::start()':
/config/esphome/build/esp32-c3/src/esphome/components/esp32_ble/ble_advertising.cpp:71: undefined reference to `esp_ble_gap_config_adv_data'
/data/cache/platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/8.4.0/../../../../riscv32-esp-elf/bin/ld: /config/esphome/build/esp32-c3/src/esphome/components/esp32_ble/ble_advertising.cpp:78: undefined reference to `esp_ble_gap_config_adv_data'
/data/cache/platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/8.4.0/../../../../riscv32-esp-elf/bin/ld: /config/esphome/build/esp32-c3/src/esphome/components/esp32_ble/ble_advertising.cpp:89: undefined reference to `esp_ble_gap_start_advertising'
collect2: error: ld returned 1 exit status
*** [/data/esp32-c3/.pioenvs/esp32-c3/firmware.elf] Error 1
========================= [FAILED] Took 65.20 seconds =========================
```


### Additional information

_No response_

--- a/esphome/components/esp32_ble/__init__.py
+++ b/esphome/components/esp32_ble/__init__.py
@@ -55,3 +55,4 @@ async def to_code(config):
 
     if CORE.using_esp_idf:
         add_idf_sdkconfig_option("CONFIG_BT_ENABLED", True)
+        add_idf_sdkconfig_option("CONFIG_BT_BLE_42_FEATURES_SUPPORTED", True)

